### PR TITLE
discord-rpc: init at 3.4.0

### PR DESCRIPTION
**Pull request by tadeokondrak**
###### Motivation for this change

Wanted to use this with ctypes.sh 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/discord-rpc/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "v${version}";
+    sha256 = "04cxhqdv5r92lrpnhxf8702a8iackdf3sfk1050z7pijbijiql2a";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ rapidjson ];
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=true"
+    "-DBUILD_EXAMPLES=${if buildExamples then "true" else "false"}"
```
Comment (aanderse) [inline pkgs/development/libraries/discord-rpc/default.nix]: `"-DBUILD_EXAMPLES=${lib.boolToString buildExamples"}` is another option.

**Review by aanderse — COMMENTED** [review]
Looking good!

**Comment by aanderse** [comment]
@GrahamcOfBorg build discord-rpc

File: pkgs/development/libraries/discord-rpc/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "-DBUILD_SHARED_LIBS=true"
+    "-DBUILD_EXAMPLES=${stdenv.lib.boolToString buildExamples}"
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Official library to interface with the Discord client";
+    homepage = "https://github.com/discordapp/discord-rpc";
+    license = licenses.mit;
+    maintainers = with maintainers; [ tadeokondrak ];
+    platforms = platforms.all;
```
Comment (aanderse) [inline pkgs/development/libraries/discord-rpc/default.nix]: Please fix build on `darwin` or update `platforms` accordingly.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/discord-rpc/default.nix (hunk 2 above)
Comment (tadeokondrak) [reply pkgs/development/libraries/discord-rpc/default.nix]: Pushed an attempt, but I don't have a Darwin machine so if it fails I'll remove it from platforms.

**Pull request merged**